### PR TITLE
Obsoleting Surface.Join Node.

### PR DESCRIPTION
**Pull request by reddyashish**
### Purpose

This PR obsoletes the Surface.Join function and suggests to use the PolySurface.ByJoinedSurfaces function. The tests are updated to use the suggested functions. 

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@aparajit-pratap @QilongTang @mjkkirschner

### CODE REVIEW

**Comment by QilongTang** [comment]
LGTM. Please make sure self CI pass and also check if there is change in DynamoRevit repo needed.

**Comment by reddyashish** [comment]
The last 2 commits have to be removed from this PR

File: test/Libraries/WorkflowTests/GeometryDefectTests.cs [hunk 1]
```diff
@@ -642,8 +642,8 @@ public void TestNodeChange_Surface_ByJoin()
             RunModel(openPath);
 
             AssertNoDummyNodes();
-            var val = GetPreviewValue("6fd810e6-45a3-43df-9303-8a0a6ce59e45") as Surface;
-            var val2 = GetPreviewValue("39df02d4-8b76-4a00-ac05-c0b7beef4db9") as Surface;
+            var val = GetPreviewValue("ac351e9f-01a4-42b3-9b49-6d764d0224b8") as Surface;
```
Comment (aparajit-pratap) [inline test/Libraries/WorkflowTests/GeometryDefectTests.cs]: What were the changes made to this DYN file? Did you remove the Surface.Join node from the graph?

Comment (reddyashish) [reply test/Libraries/WorkflowTests/GeometryDefectTests.cs]: Yes removed the surface.join node and added Polysurface.ByjoinedSurfaces node.

**Pull request merged**